### PR TITLE
Feat/#211

### LINKED ISSUE #211: Make service layer independent of persistence framework
I am currently writing an IEntityRepository implementation for MongoDB, for which there is no EF support.
The code for `Filter` and `Sort` is the same and `PageAsync` only differs in its query realization in that instead of `ToListAsync()` it is `ToAsyncEnumerable().ToList()`.

To avoid code duplication for those methods I suggest adding an abstract class that handles `Filter`, `Sort` and `PageAsync` for all DB drivers that can provide an `IQueryable<>`, with an abstract method for DB-specific query realization.
That abstract class would only depend on IJsonApiContext.

Alternatively the query realization could be in its own interface for DI.

What do you think?

### PR DESCRIPTION
**Pull request by shuebner**
Closes #211 

#### FEATURE
- [x] write tests that address the requirements outlined in the issue
- [x] fulfill the feature requirements

### CODE REVIEW

**Comment by shuebner** [comment]
Alright. I will try to fix those test failures.

**Comment by shuebner** [comment]
I added some unit tests and also took the liberty of simplifying the paging acceptance tests by reducing the amount of logic in the assert sections.

File: src/JsonApiDotNetCore/Data/DefaultEntityRepository.cs [hunk 1]
```diff
@@ -156,26 +140,36 @@ public virtual IQueryable<TEntity> Include(IQueryable<TEntity> entities, string
...
-                        .ToListAsync();
-                else // page from the end of the set                   
-                    return (await entities
-                        .OrderByDescending(t => t.Id)
-                        .Skip((Math.Abs(pageNumber) - 1) * pageSize)
-                        .Take(pageSize)
-                        .ToListAsync())
-                        .OrderBy(t => t.Id)
-                        .ToList();
+                return await entities.PageForward(pageSize, pageNumber).ToListAsync();
```
Comment (jaredcnance) [inline src/JsonApiDotNetCore/Data/DefaultEntityRepository.cs]: why are you calling the `ToListAsync()` extension method over your own `this.ToListAsync(/*...*/)` ?

Comment (shuebner) [reply src/JsonApiDotNetCore/Data/DefaultEntityRepository.cs]: No particular reason. I thought, the Mongo extension methods are an implementation detail anyway. We can use the repo's own method if you like that better.

Comment (jaredcnance) [reply src/JsonApiDotNetCore/Data/DefaultEntityRepository.cs]: That's fair, I suppose the query realization methods exposed here are really only required for the service layer's use.

**Pull request merged**